rosidl_typesupport_introspection_c/include/rosidl_typesupport_introspection_c/field_types.h

```diff
@@ -22,32 +22,54 @@ extern "C"
 {
 #endif
 
-enum
+/// Primitive (plain-old data) types supported by the ROS type system.
+enum rosidl_typesupport_introspection_c_field_types
 {
+  /// IEEE-754 binary32 format floating point number.
```

Comment:
I left the current definitions but added the design document pointer on top 19cbeb54e816cf876570e4b1f9b3fc457e12625c. I can just replace them with the corresponding C types if that makes more sense to you.

Reply:
IIUC I think the corresponding C type would be more useful because that's what users of the API should cast the member to. I won't block on this though.

Reply:
If that is going to be more useful, let's get it d82ae6047b74b5af6c2c8a01446a6bc6080b8822